newrelic/hooks/external_botocore.py

```diff
@@ -1001,6 +1002,38 @@ def _nr_sqs_message_trace_wrapper_(wrapped, instance, args, kwargs):
     return _nr_sqs_message_trace_wrapper_
 
 
+def wrap_emit_api_params(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
+    if not transaction:
+        return wrapped(*args, **kwargs)
+
+    bound_args = bind_args(wrapped, args, kwargs)
+
+    api_params = wrapped(*args, **kwargs)
+
+    arn = bound_args.get("api_params").get("FunctionName")
+    if arn and hasattr(arn, "startswith") and arn.startswith("arn:"):
+        api_params["_nr_arn"] = arn
```

Comment:
Is it possible to use a bytes string for the name? I see what you're trying to do with the `hasattr(arn, "startswith")` but that line would still crash if there's a bytes string since you can't call startswith on that and pass a string. Would it be better to just try/except this clause?